include/gsl/span

```diff
@@ -327,21 +354,24 @@ namespace details
     class extent_type<dynamic_extent>
     {
     public:
-        using index_type = std::ptrdiff_t;
+        using index_type = std::size_t;
 
         template <index_type Other>
         explicit constexpr extent_type(extent_type<Other> ext) : size_(ext.size())
         {}
 
-        explicit constexpr extent_type(index_type size) : size_(size) { Expects(size >= 0); }
+        explicit constexpr extent_type(index_type size) : size_(size)
+        {
+            Expects(size != dynamic_extent);
+        }
 
         constexpr index_type size() const noexcept { return size_; }
 
     private:
         index_type size_;
     };
 
-    template <class ElementType, std::ptrdiff_t Extent, std::ptrdiff_t Offset, std::ptrdiff_t Count>
+    template <class ElementType, std::size_t Extent, std::size_t Offset, std::size_t Count>
     struct calculate_subspan_type
     {
         using type = span<ElementType, Count != dynamic_extent
```

Comment:
Pre-existing: Consider making this an alias template, which is both simpler and higher throughput.